File: app/src/main/java/com/example/myapplication/ViewModel/MainPageViewModel.java
package com.example.myapplication.ViewModel;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import com.example.myapplication.NetworkService;
import Models.ViewModel.MainPageData;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
public class MainPageViewModel extends ViewModel {
    MutableLiveData<MainPageData> MainPage;
    public MainPageViewModel()
    {
        MainPage = new MutableLiveData<>();
        init();
    }
    public MutableLiveData<MainPageData> getMainPageData()
    {
        return MainPage;
    }
    public void init()
    {
        NetworkService.getInstance().getApi().getNewlyAdded().enqueue(new Callback<MainPageData>() {
            @Override
            public void onResponse(Call<MainPageData> call, Response<MainPageData> response) {
                MainPage.postValue(response.body());
            }
            @Override
            public void onFailure(Call<MainPageData> call, Throwable t) {

            }
        });
    }
}

File: app/src/main/java/com/example/myapplication/ViewModel/MainPageViewModel.kt
package com.example.myapplication.ViewModel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import Models.linked.Product
import com.example.myapplication.domain.usecases.*
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class MainPageViewModel(newlyAddedUseCase : GetNewlyAddedUseCase, maxDiscountedUseCase : GetMaxDiscountedUseCase,
                        mostBuyedUseCase :GetMostBuyedUseCase ) : ViewModel() {
    var NewlyAdded: MutableLiveData<ArrayList<Product>>
    var MostBuyed : MutableLiveData<ArrayList<Product>>
    var MaxDiscounted : MutableLiveData<ArrayList<Product>>
    var NewlyAddedUseCase : GetNewlyAddedUseCase
    var MostBuyedUseCase : GetMostBuyedUseCase
    var MaxDiscountedUseCase : GetMaxDiscountedUseCase
    init {
        NewlyAdded = MutableLiveData()
        MostBuyed = MutableLiveData()
        MaxDiscounted=MutableLiveData()
        NewlyAddedUseCase = newlyAddedUseCase
        MostBuyedUseCase = mostBuyedUseCase
        MaxDiscountedUseCase= maxDiscountedUseCase
    }
    fun getNewlyAdded(skip : Int, count : Int)
    {
        CoroutineScope(Dispatchers.IO).launch {
            NewlyAdded.postValue(NewlyAddedUseCase.Execute(skip, count))
        }
    }
    fun getMostBuyed()
    {
        CoroutineScope(Dispatchers.IO).launch {

        }
    }
    fun getMaxDiscounted()
    {

    }
}